src/Device/Driver/LX/NanoLogger.cpp

```diff
@@ -225,6 +225,8 @@ Nano::ReadFlightList(Port &port, RecordedFlightList &flight_list,
     env.SetProgressPosition(requested_tail - 1);
   }
 
+  std::reverse(flight_list.begin(),flight_list.end());
```

Comment:
No - there is no concern (yet) about what one specific user prefers in this code, just like it was before. It is just assumed that the user would prefer to see the most recent flight first, without having to scroll.

The stated reason is that other types of devices such as a FLARM or Butterfly or some future devices might already deliver the flights in reverse order in which case the reversal is not required, so instead of putting it into the UI affecting all drivers at the same time it is in the driver to act on LXnav devices specifically. If you prefer another solution then that is also fine of course, but this is how I am able to implement it.

Reply:
It is indeed a good idea to make it most easy for the user to select the most recent flight, but that is purely a UI thing and the driver should not make any assumptions.
The ordering of flights is an implementation detail of the logger firmware, and I don't know if any vendor has documented a guaranteed ordering of flights. This means it can change with any minor firmware update, breaking your code.
I think it really belongs in the UI and not in the driver. I'd agree to merge a PR that adds sorting to the UI, but let's keep UI things out of the driver code.

Reply:
I only know the LXNav devices, I never got FLARM to work with my hardware, and never used other types of loggers. Which order do other drivers provide? Is it true that all of them deliver oldest flight first? That is the relevant question.  (And in that case UI is the better place)

(At any rate I will not be able to do the UI thing for the foreseeable future, I will keep my implementation for my own flights)

Reply:
> (At any rate I will not be able to do the UI thing for the foreseeable future, I will keep my implementation for my own flights)

Actually, it is not so bad. The same code can be applied on the UI level in the dialog code as it uses still the same data type. I have that commit, it compiles fine. It is the same line of code in a different file. So, now it is implemented in the UI, I just hope that users of other logger types now do not complain about suddenly having oldest flight first.

So that is fine, however I am not able to test this version with an actual device. The same code in the driver was tested with an actual device. Of course I have tested it even less with other types of devices, which are now all affected. If you consider my previous test good enough, then use the commit. Otherwise I remove it from the PR until I have tested my own XC version, which includes it.

Reply:
No, that's bad. You can't just reverse the order. The order of flights from the logger (or from the driver) is undefined. If stuff works now, it is by coincidence, but you must not break that. You can replace it with a well-defined order, but not randomly with something else.
The right function you need is `std::sort()`, not `std::reverse()`.